### PR TITLE
refactor: move to taskfile

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments

This is a refactor to move from invoke to taskfiles.

❗️This may not work on initial commit because the submodule will need to be updated to point to the latest release of the goat.

**Important notes**

1. Note that this removes the `run_security_tests` which used to run trivy but would exit 0 regardless and wasn't really operationally reviewed. We can re-add it in if we are going to actually leverage the security scanning information (consider syft/grype instead, etc.)
2. There is a lot of refactoring of how vars are passed around, which supports more concise variable usage.
3. This now leverages the "base" Taskfile.yml much more, and has adjustments for bash vs python. This was an expected follow-on to the initial python/Taskfile.yml.
4. Instead of replacing `pipenv run invoke goat` with `task goat` we are now running `task test`, which is more standard.
5. Please test by pointing a Python project’s submodule to this branch

## Testing

This is a significant change and should be manually tested. Please:

1. Clone this branch, follow the `README.md` instructions, and run some `task` commands. I especially recommend simulating the GitHub Actions as those are more difficult to test.
2. Generate a paved road project, update the submodule to point to this branch, pull the changes down, and run the base commands to ensure it works. However, this will not go into effect until downstream projects update their submodule, which is by design so if there's a bug it won't break all of our builds 😀

## Pull Request Checklist

Thank you for submitting a contribution to the goat!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, link the issue to the PR.

### CODE REVIEW

File: .github/CONTRIBUTING.md
Comment (JonZeolla) [inline .github/CONTRIBUTING.md:91]: Need to test

File: .github/workflows/commit.yml
Comment (JonZeolla) [inline .github/workflows/commit.yml]: We may need to update clone instructions to include `--submodules`

File: Task/Taskfile.yml
Comment (JonZeolla) [inline Task/Taskfile.yml:41]: This is new; required so we can pass the vars around more directly instead of doing global vars

File: Task/Taskfile.yml
Comment (JonZeolla) [inline Task/Taskfile.yml:68]: This is new; uses the `default` function and will drop to "UNKNOWN" but otherwise takes the provided `VERSION` and uses it.  Downstream (`bash/Taskfile.yml` and `python/`) should be passing a version with logic specific to bash/python.

File: Task/Taskfile.yml
Comment (JonZeolla) [inline Task/Taskfile.yml]: I added the `{{.ROOT_DIR}}` part of this so it has the right context

File: Task/Taskfile.yml
Comment (JonZeolla) [inline Task/Taskfile.yml:148]: This is a brand new `task` primarily to fix semver releases or enable latest-only releases for a subset of calver projects.

File: .github/CONTRIBUTING.md
Comment (JonZeolla) [inline .github/CONTRIBUTING.md:6]: Don't forget to repoint the submodule to `main`

Comment (JonZeolla) [reply .github/CONTRIBUTING.md:6]: I am suggesting we purposefully don't do this so that this merge succeeds, and then after CI/CD completes we repoint it back to `main`.

File: Dockerfile
Comment (JonZeolla) [inline Dockerfile:1]: The `platform` work here and in the `Taskfile.yml` files was necessary for my local testing/confirmation that the goat works.

File: Task/Taskfile.yml
Comment (JonZeolla) [inline Task/Taskfile.yml:6]: This essentially runs `set -euo pipefail` which we always want.  Same with the other `Taskfile.yml`s

File: Task/Taskfile.yml
Comment (JonZeolla) [inline Task/Taskfile.yml:50]: Once we build **and push** arm64 images, then we can remove this.

Comment (JonZeolla) [reply Task/Taskfile.yml:50]: I self-assigned a task for next sprint - https://www.wrike.com/open.htm?id=1167647127

File: .github/CONTRIBUTING.md
Comment (JonZeolla) [reply .github/CONTRIBUTING.md:6]: I have a task for next sprint to fix this - https://www.wrike.com/open.htm?id=1167647370

It is blocked on another task - https://www.wrike.com/open.htm?id=1113530084 - to update the pipelines to build and push multiplatform images

File: Dockerfile
Comment (JonZeolla) [reply Dockerfile:1]: See more info in https://docs.docker.com/build/building/multi-platform/#building-multi-platform-images and https://taskfile.dev/usage/#platform-specific-tasks-and-commands

File: README.md
Comment (JonZeolla) [inline README.md:124]: I cleaned this up to only the log levels we actually use and added debug logging because I needed it to troubleshoot some changes that I was making.

File: entrypoint.sh
Comment (JonZeolla) [inline entrypoint.sh]: This is needed so we can do the more explicit `^/{{.PROJECT_SLUG}}/goat/.*` exclusion

File: VERSION
Comment (JonZeolla) [inline VERSION:1]: This is a new required file for bash-in-docker projects for managing versions

**Comment by JonZeolla** [comment]
The build failure is just a bootstrapping issue, I can push a tag later

**Comment by JonZeolla** [comment]
CodeQL just needs a config change too because of removing tasks.py

**Comment by JonZeolla** [comment]
May need to add some bumpversion logic as well (run `task release` in the pipeline?)

File: .github/workflows/commit.yml
Comment (JonZeolla) [reply .github/workflows/commit.yml]: I don’t think we need to

**Comment by JonZeolla** [comment]
Ok the last failures have to do with `jscpd` not properly accounting for the `INPUT_EXCLUDE` variable.  Any thoughts @ChiefHolland ?

One idea is we could append the `INPUT_EXCLUDE` to the `JSCPD_CONFIG` but they use different regex syntax 🤔

**Comment by ChiefHolland** [comment]
> Ok the last failures have to do with `jscpd` not properly accounting for the `INPUT_EXCLUDE` variable.  Any thoughts @ChiefHolland ?
> 
> 
> 
> One idea is we could append the `INPUT_EXCLUDE` to the `JSCPD_CONFIG` but they use different regex syntax 🤔

Most likely due to jscpd file type in linters.json being set to "all". I'll double check the logic for that case.

File: .github/CONTRIBUTING.md
Comment (JonZeolla) [reply .github/CONTRIBUTING.md:91]: I tested and this works

File: .github/workflows/commit.yml
Comment (JonZeolla) [reply .github/workflows/commit.yml]: We don't; confirmed this works fine with just `Taskfile.yml` updates

File: Task/Taskfile.yml
Comment (JonZeolla) [reply Task/Taskfile.yml:41]: However, note that we do use global vars when they make sense to reduce duplication or excessive passing around of vars

File: .gitmodules
Comment (JonZeolla) [inline .gitmodules:4]: We need to undo this after this merges; there's a follow-on task for that

File: .github/CONTRIBUTING.md
Comment (ChiefHolland) [inline .github/CONTRIBUTING.md]:
```suggestion
    To see more information, add debug messages to output with `task test -- debug`  or enable verbose mode `task -v test`.
```

File: README.md
Comment (ChiefHolland) [inline README.md]:
```suggestion
than other open source multi-linter projects, and easily extended to enforce Seiso policies programmatically.
```

Comment (ChiefHolland) [reply README.md]: nit: split infinitive

**Review by ChiefHolland — COMMENTED** [review]
Feedback in comments.

File: .github/CONTRIBUTING.md
Comment (JonZeolla) [reply .github/CONTRIBUTING.md]: Ah that was just at typo let me fix it.

File: README.md
Comment (JonZeolla) [reply README.md]: Done!

File: .github/CONTRIBUTING.md
Comment (ChiefHolland) [inline .github/CONTRIBUTING.md]:
```suggestion
    Or run `task test -- debug` to run it in debug mode.
```

**Review by ChiefHolland — APPROVED** [review]
LGTM

**Pull request merged**
